pyfar/io/io.py

```diff
@@ -461,6 +463,320 @@ def default_audio_subtype(format):
     return soundfile.default_subtype(format)
 
 
+def read_comsol(filename, expressions=None, parameters=None):
+    """Read exported data from COMSOL.
+
+    Parameters
+    ----------
+    filename : str, Path
+        Input file. Excepted input files are '.txt', '.dat' and '.csv'. '.csv'
+        Files are strongly recommended, since .txt and .dat are inconvenient
+        in their format.
+    expressions : list[str], optional
+        This list of expressions defines the output expressions. An example
+        can be like ``expressions=['pabe.Lp_t']``. Also see
+        :py:func:`~pyfar.dsp.comsol_read_header`. By default, all expressions
+        get returned.
+    parameters : dict
+        This dict contains the parameters, which sould be read. An example can
+        be like ``parameters={'theta': [0.0, 0.7854], 'phi': [0., 1.5708]}``.
+        Also see :py:func:`~pyfar.dsp.comsol_read_header`. Default is None,
+        then all parameters are read.
+
+    Returns
+    -------
+    data : Signal or FrequencyData
+        Returns a Signal or FrequencyData object depending on the input data.
+    coordinates : Coordinates, optional
+        A Coordinates Object based on the Input Data. If the input dimension is
+        lower than three, the missing dimensions are set to zero. If there are
+        no positions given, this parameter will not be returned.
+
+    Raises
+    ------
+    SyntaxError
+        Raises SyntaxError if the file format is not supported.
+    ValueError
+        Raises ValueError if the Frequency or Time data are inconvenient in the
+        input file.
+
+    Examples
+    --------
+    Returns all expressions
+    >>> import pyfar as pf
+    >>> data, coordinates = pf.io.read_comsol('my_comsol_data.csv')
+    """
+
+    # check Datatype
+    suffix = pathlib.Path(filename).suffix
+    if suffix not in ['.txt', '.dat', '.csv']:
+        raise ValueError((
+            "Input path must be a .txt, .csv or .dat file"
+            f"but is of type {str(suffix)}"))
+
+    # get header
+    all_expressions, units, all_parameters, domain, domain_data \
+        = read_comsol_header(filename)
+    if 'dB' in units:
+        warnings.warn(r'Do you really want to use dB-values?')
```

Comment:
I would suggest a more verbose warning that tells the user why this might be risky. Maybe something like 'The data contains values in dB. Consider to de-logarithmize the data if possible. Otherwise any further processing of the data might lead to erroneous results.'